components/board/genericlinux/data.go

```diff
@@ -96,16 +98,14 @@ func GetGPIOBoardMappings(modelName string, boardInfoMappings map[string]BoardIn
 // getCompatiblePinDefs returns a list of pin definitions, from the first BoardInformation struct
 // that appears compatible with the machine we're running on.
 func getCompatiblePinDefs(modelName string, boardInfoMappings map[string]BoardInformation) ([]PinDefinition, error) {
-	const compatiblePath = "/proc/device-tree/compatible"
+	var compatibles utils.StringSet
+	var err error
+
+	compatibles, err = rdkutils.GetArchitectureInfo(modelName)
 
-	compatiblesRd, err := os.ReadFile(compatiblePath)
 	if err != nil {
-		if os.IsNotExist(err) {
-			return nil, noBoardError(modelName)
-		}
-		return nil, err
+		return nil, fmt.Errorf("error while getting hardware info %w", err)
```

Comment:
It's not clear to me that the `fmt.Errorf(...)` adds any new information. Why not just `return nil, err`? (perhaps it does add something important, and I didn't notice.)

```go
compatibles, err := rdkutils.GetArchitectureInfo(modelname)
if err != nil {
    return nil, err
}
```

Reply:
added additional comments for better trackback, since a lot of errors in RDK is hard to trace.